azure-pipelines.yaml

```diff
@@ -0,0 +1,121 @@
+trigger:
+  - master
+pr:
+  - none
+
+parameters:
+  - name: Run_Stage
+    displayName: Stage to Run
+    type: string
+    default: 'Deploy_Agent'
+    values:
+    - Deploy_Agent
+    - Destroy_Agent
+
+  - name: environment
+    displayName: Environment (sandbox default)
+    type: string
+    default: sandbox # USE upper and lower
+    values:
+      - sandbox
+ #     - upper
+ #     - lower
+
+variables:
+  - group: vh-azure-devops-agent
+  - name: TF_VAR_RG_NAME
+    value: "vh-devops-agent-rg-${{ parameters.environment }}"
+  - name: TF_VAR_KV_NAME
+    value: "vhdoagentkv${{ parameters.environment }}"
+  - name: TF_VAR_VM_NAME
+    value: "vh-devops-agent-vm-${{ parameters.environment }}"
+  - name: TF_VAR_NSG_NAME
+    value: "vh-devops-vm-nsg-${{ parameters.environment }}"
+  - name: TF_VAR_VNET_NAME
+    value: "vh-devops-vnet-${{ parameters.environment }}"
+
+stages:
+  
+  - stage: Deploy_Agent
+    condition: eq('${{ parameters.Run_Stage }}', 'Deploy_Agent')
+    displayName: Deploy_Agent
+    jobs:
+      - job: Deploy_Terraform
+        displayName: Deploy Terraform
+        pool:
+          vmImage: 'ubuntu-latest'
+        
+        steps:
+
+          - template: pipeline-steps/replace-token.yaml
+            parameters:
+              myConnection: $(patToken)
+              workingDirectory: $(System.DefaultWorkingDirectory)/$(tfDirectory)
+              environment: ${{ parameters.environment }}
+
+              
+          - template: pipeline-steps/terraform-prep.yaml
+            parameters:
+              terraformVersion: $(tfVersion)
+              subscription: $(subscription)
+              environment: ${{ parameters.environment }}
+              resourceGroup: $(resourceGroup)
+              location: $(location)
+              storageAccount: $(storageAccount)
+              containerName: $(containerName)
+              stateKey: $(stateKey)-${{ parameters.environment }}.tfstate
+              workingDirectory: $(System.DefaultWorkingDirectory)/$(tfDirectory)
+
+          - template: pipeline-steps/terraform-plan.yaml
+            parameters:
+              subscription: $(subscription)
+              environment: ${{ parameters.environment }}
+              tfVariables: $(System.DefaultWorkingDirectory)/$(tfDirectory)/$(tfVariables)
+              workingDirectory: $(System.DefaultWorkingDirectory)/$(tfDirectory)
+
+          - template: pipeline-steps/terraform-apply.yaml
+            parameters:
+              subscription: $(subscription)
+              environment: ${{ parameters.environment }}
+              planLocation: $(System.DefaultWorkingDirectory)/$(tfDirectory)/${{ parameters.environment }}.tfplan
+              workingDirectory: $(System.DefaultWorkingDirectory)/$(tfDirectory)
+
+# If you need to destroy this infra, delete and purge the secret from the keyvault before you run the detroy stage
```

Comment:
Can we do this instead via the pipeline?
https://docs.microsoft.com/en-us/cli/azure/keyvault/secret?view=azure-cli-latest#az-keyvault-secret-delete
https://docs.microsoft.com/en-us/cli/azure/keyvault/secret?view=azure-cli-latest#az-keyvault-secret-purge